client/python/djclient/dj.py

```diff
@@ -63,24 +64,174 @@ def construct_url(self, url):
+        display_name: Optional[str] = None,
+        primary_key: Optional[List[str]] = None,
+        tags: Optional[List["Tag"]] = None,
+    ) -> "Transform":
+        """
+        Instantiates a new Transform object with the given parameters.
+        """
+        return Transform(
+            dj_client=self,
+            name=name,
+            description=description,
+            display_name=display_name,
+            tags=tags,
+            primary_key=primary_key,
+            query=query,
+        )
+
+    def dimension(self, node_name: str) -> "Dimension":
+        """
+        Retrieves a dimension node with that name if one exists.
+        """
+        node_dict = self.verify_node_exists(node_name, "dimension")
+        return Dimension(
+            **node_dict,
+            dj_client=self,
+        )
+
+    def new_dimension(  # pylint: disable=too-many-arguments
+        self,
+        name: str,
+        query: str,
+        primary_key: Optional[List[str]],
+        description: Optional[str] = None,
+        display_name: Optional[str] = None,
+        tags: Optional[List["Tag"]] = None,
+    ) -> "Dimension":
+        """
+        Instantiates a new Dimension object with the given parameters.
+        """
+        return Dimension(
+            dj_client=self,
+            name=name,
+            description=description,
+            display_name=display_name,
+            tags=tags,
+            primary_key=primary_key,
+            query=query,
+        )
+
+    def metric(self, node_name: str) -> "Metric":
+        """
+        Retrieves a metric node with that name if one exists.
+        """
+        node_dict = self.verify_node_exists(node_name, "metric")
+        return Metric(
+            **node_dict,
+            dj_client=self,
+        )
+
+    def new_metric(  # pylint: disable=too-many-arguments
+        self,
+        name: str,
+        query: str,
+        description: Optional[str] = None,
+        display_name: Optional[str] = None,
+        primary_key: Optional[List[str]] = None,
+        tags: Optional[List["Tag"]] = None,
+    ) -> "Metric":
+        """
+        Instantiates a new Metric object with the given parameters.
+        """
+        return Metric(
+            dj_client=self,
+            name=name,
+            description=description,
+            display_name=display_name,
+            tags=tags,
+            primary_key=primary_key,
+            query=query,
+        )
+
+    def verify_node_exists(self, node_name: str, type_: str) -> Dict[str, Any]:
+        """
+        Retrieves a node and verifies that it exists and has the expected node type.
+        """
+        node = self.get_node(node_name)
+        if "name" not in node:
+            raise DJClientException(f"No node with name {node_name} exists!")
+        if "name" in node and node["type"] != type_:
+            raise DJClientException(
+                f"A node with name {node_name} exists, but it is not a {type_} node!",
```

Comment:
nit: no shouting at the users 😄 
![image](https://user-images.githubusercontent.com/43911210/232863117-86402385-e7ae-41cc-97f5-174f49c13d38.png)

Reply:
Hahaha oops... that should be an April Fools release, where all error messages are in all-caps.